### PR TITLE
Extend radio button for checkbox

### PR DESCRIPTION
**Pull request by mlunoe**
No change in functionality, just reuse of code from radio button.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @rcorral

File: src/Form/__tests__/FieldCheckbox-test.js [hunk 1]
```diff
@@ -95,33 +97,200 @@ describe('FieldCheckbox', function () {
...
+
+      var inputComponent = TestUtils.findRenderedDOMComponentWithTag(
+        instance, 'input'
+      );
+      var el = ReactDOM.findDOMNode(inputComponent);
+
+      expect(el.checked).toEqual(false);
+    });
+
+    it('is not checked by default', function () {
```
Comment (jfurrow) [inline src/Form/__tests__/FieldCheckbox-test.js]: This description should be something like `allows setting initial checked value`

**Comment by kennyt** [comment]
hey i tested this on the rjsc docs and for the single checkboxes, the value becomes an object. previously, the functionality returned true or false

File: src/Form/FieldCheckbox.js [hunk 2]
```diff
@@ -1,104 +1,109 @@
...
+      let checkbox = ReactDOM.findDOMNode(refs[refName]);
+      let indeterminate;
+      // Single checkbox
+      if (props.name === refName) {
+        indeterminate = props.indeterminate;
+      }
 
-    return errorMsg;
+      // Multiple checkboxes
+      if (Util.isArray(props.startValue)) {
```
Comment (MatApple) [inline src/Form/FieldCheckbox.js]: I think we should use `Array.isArray` now that we have it natively :)

File: src/Form/FieldCheckbox.js [hunk 3]
```diff
@@ -1,104 +1,109 @@
...
+        indeterminate = Util.find(props.startValue, function (item) {
```
Comment (MatApple) [inline src/Form/FieldCheckbox.js]: Can also use native find now!

File: src/Form/FieldCheckbox.js [hunk 4]
```diff
@@ -1,104 +1,109 @@
...
 
 FieldCheckbox.propTypes = {
-  // Optional value for setting the checked state of the checkbox,
-  // should be either true for 'checked' or false 'unchecked'
-  checked: React.PropTypes.bool,
   // Optional number of columns to take up of the grid
   columnWidth: React.PropTypes.number.isRequired,
+  // Function to handle change event
+  // (usually passed down from form definition)
+  handleEvent: React.PropTypes.func,
```
Comment (MatApple) [inline src/Form/FieldCheckbox.js]: This should be a required type right?

File: src/Form/__tests__/FieldCheckbox-test.js [hunk 5]
```diff
@@ -95,33 +97,200 @@ describe('FieldCheckbox', function () {
...
+        IconCheckbox
       )).toEqual(true);
     });
 
+    it('should display items pass in from startValue', function () {
+      expect(this.children.length).toEqual(3);
+    });
+
+    it('should display parent labelClass on all items', function () {
+      for (var i = 0; i < this.children.length; i++) {
```
Comment (MatApple) [inline src/Form/__tests__/FieldCheckbox-test.js]: var

File: src/Form/__tests__/FieldCheckbox-test.js [hunk 6]
```diff
@@ -95,33 +97,200 @@ describe('FieldCheckbox', function () {
...
+        expect(this.children[i].props.labelClass).toEqual('foo');
+      }
+    });
+
+    it('should override parent labelClass with item labelClass', function () {
+      var instance = TestUtils.renderIntoDocument(
```
Comment (MatApple) [inline src/Form/__tests__/FieldCheckbox-test.js]: Let change to `let` :) throughout this test file

File: src/Form/icons/IconCheckbox.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  React.PropTypes.object,
+  React.PropTypes.string
+]);
+
+IconCheckbox.propTypes = {
+  checked: React.PropTypes.bool,
+  indeterminate: React.PropTypes.bool,
+
+  // Classes
+  className: classPropType
```
Comment (MatApple) [inline src/Form/icons/IconCheckbox.js]: 🔤

File: src/Form/icons/IconCheckbox.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        <g className={checkIconClasses}>
+          <path d="M6.37547375,12.4433526 C6.66651906,12.8810343 7.30583183,12.8913338 7.61082475,12.4632544 L12.0274757,6.26416941 C12.2678262,5.92682021 12.1891933,5.45850202 11.8518441,5.21815145 C11.5144949,4.97780088 11.0461767,5.0564338 10.8058262,5.393783 L7.02648971,10.7134266 L5.62452625,8.60511675 C5.39516713,8.26020042 4.92962491,8.1665228 4.58470858,8.39588191 C4.23979226,8.62524103 4.14611463,9.09078326 4.37547375,9.43569958 L6.37547375,12.4433526 Z" fill="#000000" fillOpacity="0.13" />
+          <path d="M6.37547375,11.4433526 C6.66651906,11.8810343 7.30583183,11.8913338 7.61082475,11.4632544 L12.0274757,5.26416941 C12.2678262,4.92682021 12.1891933,4.45850202 11.8518441,4.21815145 C11.5144949,3.97780088 11.0461767,4.0564338 10.8058262,4.393783 L7.02648971,9.71342659 L5.62452625,7.60511675 C5.39516713,7.26020042 4.92962491,7.1665228 4.58470858,7.39588191 C4.23979226,7.62524103 4.14611463,8.09078326 4.37547375,8.43569958 L6.37547375,11.4433526 Z" fill="#ffffff" />
+        </g>
+      </svg>
+    );
+  }
+}
+
+let classPropType = React.PropTypes.oneOfType([
```
Comment (MatApple) [inline src/Form/icons/IconCheckbox.js]: #nit Seeing we have a lot of proptype declarations we could import `PropTypes` at the top with `import React, {PropTypes} from 'react';`

File: src/Form/FieldCheckbox.js (hunk 4 above)
Comment (mlunoe) [inline src/Form/FieldCheckbox.js]: Done

**Pull request merged**